src/MongoDB.Driver/MongoUrlBuilder.cs

```diff
@@ -747,6 +753,7 @@ public void Parse(string url)
             else if (connectionString.WaitQueueMultiple != null)
             {
                 _waitQueueMultiple = connectionString.WaitQueueMultiple.Value;
+#pragma warning restore 618
```

Comment:
It seems a little unbalanced to restore the warning in the middle of the if statement.

Should we try to use #pragmas around whole statements (even if they are compound statements like if)?

If so that would mean moving the restore to after line 758.